gluon-mesh-vpn-wireguard-vxlan/files/lib/gluon/upgrade/400-mesh-vpn-wireguard

```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+
+# Make sure we store the private key
+privkey=$(uci get wireguard.wireguard.privatekey)
+
+echo > /etc/config/wireguard
+
+uci set wireguard.wireguard=wireguard
+
+
+#Are we running in multidomain mode?
+domain=$(uci get gluon.core.domain)
+if [ -z "$domain" ]; then {                     #nope
+        datei="/lib/gluon/site.json";
+} else {                                        #yep
+        datei="/lib/gluon/domains/$domain.json";
+}
+fi 
+
+uci set wireguard.wireguard.enabled=$(jsonfilter -i $datei -e "$.mesh_vpn.wireguard.enabled")
+uci set wireguard.wireguard.iface=$(jsonfilter -i $datei  -e "$.mesh_vpn.wireguard.iface")   
+uci set wireguard.wireguard.limit=$(jsonfilter -i $datei -e "$.mesh_vpn.wireguard.limit")     
+uci set wireguard.wireguard.privatekey=$privkey
```

Comment:
Again, I don't know the context this is running in or whether it's a multi user system, but this leaks the private key into the process table. e.g. tools like `ps` can see, momentarily, the private key while that command is running.

Reply:
Runs on OpenWrt in a single-user system. Ideally this whole script could be migrated to Lua.

Reply:
Ah. Well, you can still work around it if you want with

`echo "set wireguard.wireguard.privatekey=$privkey" | uci batch`

Since `echo` is an ash built-in and thus won't leak to the process table.

Reply:
I added all your suggestions many thanks @zx2c4